GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -557,6 +573,26 @@ private void ResetSelectedItemsTo(bool actsAsChild, bool deleteUncommittedAddedI
             }
         }
 
+        /// <summary>
+        /// It is generally not possible to reset to an artificial commit.
+        /// </summary>
+        /// <param name="resetId">The selected commit it.</param>
+        /// <returns><see langword="null"/> if not possible to reset to this revision.</returns>
+        private ObjectId? GetResetId(ObjectId? resetId)
+        {
+            return (resetId?.IsArtificial is false) ? resetId : null;
+        }
```

Comment:
```suggestion
        private bool CanReset(ObjectId? resetId, ObjectId? secondId)
        {
            return (resetId?.IsArtificial is false) || (resetId == ObjectId.IndexId && secondId == ObjectId.WorkTreeId);
        }
```

Reply:
It is possible to multi select, so the check has to be a little more complicated. Something like below.
`IsParentIndex()` will run slightly more often, we can probably ignore that OH though (even if I avoided it above).
Will think about it, not sure I like it.

I want to have a rest-to-head in revdiff too. The original proposal limited to Second for Index->WorkTree would not be so bad, but not in this PR. Need to finish the 4.1stuff, I guess you want reviews on graph changes?

```
private bool CanResetToSecond(ObjectId? resetId)
        {
            return resetId?.IsArtificial is false;
        }
private bool CanResetToFirst(ObjectId? resetId, IEnumerable<FileStatusItem> selectedItems)
        {
            return CanResetToSecond(resetId) || (selectedItems.SecondIds().All(i => i == ObjectId.WorkTreeId) && selectedItems.FirstIds().All(i => i == ObjectId.IndexId));
        }
```

Reply:
Pushed to tmp/i10899-diff-reset-head-2 for now

Reply:
> It is possible to multi select

Yes. (I mixed this with the FileStatusList where all files have the same first & second commit. And below, only the first selection is evaluated.)

> Need to finish the 4.1stuff

Me too. 4.1.1 is for bugfixes only.

Reply:
I want to add the reset fixes to 4.1.1, this is a prerequisite

Reply:
Used tmp/i10899-diff-reset-head-2 with suggested changes and doc cleanup